pkg/segment/results/mresults/metricresults.go

```diff
@@ -389,7 +389,7 @@ func (res *MetricsResult) GetMetricTagsResultSet(mQuery *structs.MetricsQuery) (
 		}
 	}
 
-	tagKeyValueSet := make([]string, 0)
+	tagKeyValueSet := make(map[string]bool)
```

Comment:
You never need these `bool` values, right? If not, I think it would be clearer to make this map to empty structs. Then the type could be `map[string]struct{}` and you set values like `myMap["apple"] = struct{}{}`

Reply:
ah  I like this!